### DISCUSSION TITLE
Action "Tests" failed for "WindowsPowerShell" task

### DISCUSSION THREAD

**Discussion by amorphobia** [ScoopInstaller/Scoop#5042] [General]
Recent days, the CI action "Tests" failed for the same reason:

```
  Describing Manifest Validator
    [-] Error occurred in Describe block 0ms
      ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
      at <ScriptBlock>, D:\a\Main\Main\scoop_core\test\Import-Bucket-Tests.ps1: line 35
      at Invoke-Blocks, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 135
      at Invoke-TestGroupSetupBlocks, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 121
      at DescribeImpl, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\Describe.ps1: line 209
  Describing manifest validates against the schema
    [-] Error occurred in Describe block 0ms
      PSArgumentException: Cannot find type [Scoop.Validator]: verify that the assembly containing this type is loaded.
      at <ScriptBlock>, D:\a\Main\Main\scoop_core\test\Import-Bucket-Tests.ps1: line 86
      at Invoke-Blocks, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 135
      at Invoke-TestGroupSetupBlocks, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 121
      at DescribeImpl, C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Pester\4.10.1\Functions\Describe.ps1: line 209
Tests completed in 4.57s
Tests Passed: 6, Failed: 2, Skipped: 0, Pending: 0, Inconclusive: 0 
Error: Process completed with exit code 1.
```

Is it a bug? Any ideas?

**Comment by rashil2000** [comment]
Yes, it is known https://github.com/ScoopInstaller/Scoop/pull/5026#issuecomment-1179649455

**Reply by amorphobia** [reply]
Thanks

**Reply by ViCrack** [reply]
I have the same problem

**Comment by niheaven** [comment]
Fixed in #5053, and will be in next release.

**Comment by Xelbayria** [comment]
Hello, let me know if I need to open a new discussion regarding the problem. Ok, so I've created my own bucket with a list of packages that doesn't exist in other buckets & have problem with "TEST" via github's action where it is testing the "bucket". The error are below:  

the name is "Test Bucket" 
```
Error: [-] files end with a newline 133ms (130ms|3ms)
Message
  RuntimeException: The following files do not end with a newline: 
  D:\a\utilities\utilities\my_bucket\bucket\autotag.json
```
and there are a few more errors: 
- ` Error: [-] files have no lines containing trailing whitespace 22ms (20ms|1ms)`
- ` Error: [-] any leading whitespace consists only of spaces (excepting makefiles) 16ms (15ms|1ms)`
- `Error: Process completed with exit code 1.`

Is there a way I can fix this?

**Reply by HUMORCE** [reply]
> Error: [-] files end with a newline 133ms (130ms|3ms)

Insert a newline to end of file.

> Error: [-] files have no lines containing trailing whitespace 22ms (20ms|1ms)

Remove trailing whitespace for everyline.

> Error: [-] any leading whitespace consists only of spaces (excepting makefiles) 16ms (15ms|1ms)

Use `space` indents instead of tab

**Reply by Xelbayria** [reply]
ah of course. the issue was the formation of JSON for autotag.json. Is there a way I can locally use "test.ps1" to run test on the file to make sure the formation is correct? I supposed that test.ps1 should be similar to `/test` in PR

**Reply by HUMORCE** [reply]
https://github.com/ScoopInstaller/BucketTemplate/blob/master/.editorconfig

**Reply by Xelbayria** [reply]
I see. that's how it works. I'll assume that it's not possible to run `test.ps1` locally on one file on my computer.

**Reply by niheaven** [reply]
On one file? You could run test locally (test for whole bucket) by `./bin/test/ps1`.

**Reply by Xelbayria** [reply]
> On one file? You could run test locally (test for whole bucket) by `./bin/test/ps1`.

I believe you meant to say `./bin/test.ps1`. To be honest, I've already tried that, it don't work because it's missing "BuildHelper"

```
./bin/test.ps1: The script 'test.ps1' cannot be run because the following modules that are specified by the "#requires" statements of the script are missing: BuildHelpers.
```
that's why I asked the question, so yes I want to run test on one file to make sure the formation are all good.

**Reply by HUMORCE** [reply]
> To be honest, I've already tried that, it don't work because it's missing "BuildHelper"

```
❯ cat "$(scoop prefix scoop)\test\bin\test.ps1" | Select-String '#requires'

#Requires -Version 5.1
#Requires -Modules @{ ModuleName = 'BuildHelpers'; ModuleVersion = '2.0.1' }
#Requires -Modules @{ ModuleName = 'Pester'; ModuleVersion = '5.2.0' }
#Requires -Modules @{ ModuleName = 'PSScriptAnalyzer'; ModuleVersion = '1.17.1' }
```

Install these modules first.

**Comment by Xelbayria** [comment]
> > To be honest, I've already tried that, it don't work because it's missing "BuildHelper"
> 
> ```
> ❯ cat "$(scoop prefix scoop)\test\bin\test.ps1" | Select-String '#requires'
> 
> #Requires -Version 5.1
> #Requires -Modules @{ ModuleName = 'BuildHelpers'; ModuleVersion = '2.0.1' }
> #Requires -Modules @{ ModuleName = 'Pester'; ModuleVersion = '5.2.0' }
> #Requires -Modules @{ ModuleName = 'PSScriptAnalyzer'; ModuleVersion = '1.17.1' }
> ```
> 
> Install these modules first.

Ok, so which source do I use? I tried to install them via Powershell 5.1 and these modules like BuildHelper do not exist.

**Reply by HUMORCE** [reply]
- https://www.powershellgallery.com/packages/BuildHelpers/2.0.1
- https://www.powershellgallery.com/packages/Pester/5.2.0
- https://www.powershellgallery.com/packages/PSScriptAnalyzer/1.17.1

**Reply by Xelbayria** [reply]
Thank you! i got the `./bin/test.ps1` working and i got one error on one file, gotta fix it. have a great day
